dali/base/dafdesc.cpp

```diff
@@ -1353,14 +1357,29 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
                 *_interleaved = interleaved;
         }
 
-        const char * historyStr = "History";
-        if (mb.remaining() > strlen(historyStr))
+        // If we have more serialised data in MemoryBuffer
+        if (mb.remaining() >= sizeof(unsigned short))
         {
-            const char * buffer = mb.toByteArray()+mb.getPos();
-            if (nullptr != strstr(buffer, historyStr))
-                history.setown(createPTree(mb));
+            unsigned short extendedVersion;
+            mb.read(extendedVersion);
+
+            if (extendedVersion >= VERSION_MINHISTORY)
+            {
+                //If extendedVersion >= 1 then we should have stored History
+                const char * historyStr = "History";
+                if (mb.remaining() > strlen(historyStr))
+                {
+                    const char * buffer = mb.toByteArray()+mb.getPos();
+                    if (nullptr != strstr(buffer, historyStr))
+                        history.setown(createPTree(mb));
+                    else
+                        history.setown(createPTree(historyStr));
```

Comment:
Link works for me.
Remade the point above anyway.

You shouldn't be manually checking (strstr etc), just deserialize it. You have already validated the version. So:
```
if (extendedVersion >= VERSION_MINHISTORY)
    history.setown(createPTree(historyStr));
else
    throw ...
```